upload.php

```diff
@@ -97,6 +95,11 @@
     $bm = new booking_manager($f);
     $bm->load_from_file($fileid);
 
+    // Get the options selected by the user at confirm time.
+    $confirmdata = (new confirm_bookings_form(null))->get_data();
+    $suppressemail = !empty($confirmdata->suppressemail);
+    $bm->set_suppress_email($suppressemail);
```

Comment:
I wonder if it's generally a good idea to have these setters `set_suppress_email($bool)`, or to have an explicit `$bm->suppress_emails()` method instead, and have the caller control the intended behaviour, but the class control the value being set. 

Wondering in terms of debugging, since it defaults (in the bm class) to send emails, the only times it shouldn't is when it hits this `suppress_emails` call, which can be more clearly observed if viewed via a stack trace for example, instead of needing to see it hit this setter, and checking if the value passed to this setter was indeed true/false and correctly set (or something to that nature).

Reply:
I think both ways are more or less equal, with my deciding factor being simplicity of `->suppress_emails()` without any var, just setting it to true. Have updated the PR accordingly.